File: hts/strings/strings.py


primes_dict = [2,3,5,7]

fd = open('data.txt', 'r')
string = fd.read()
string = string.strip()
primes = []
comps = []
alpha = []

for c in string:
    try:
        num = int(c)
        if num != 0 and num != 1:
            if num in primes_dict:
                primes.extend([num])
            else:
                comps.extend([num])
    except ValueError:
        if len(alpha) < 25:
            alpha.extend([c])

numerics = sum(primes)*sum(comps)
alpha = [ chr(ord(x)+1) for x in alpha ]
print( ''.join(alpha)+str(numerics) )

File: hts/strings/strings.js


var primes_dict = [2,3,5,7];
var string = $("input").first().val();
var primes = [];
var comps = [];
var alpha = [];

for( var i = 0; i < string.length; i++ ){
    var c = string[i];
    if(parseInt(c)){
	c = parseInt(c)
	if( c != 0 && c != 1 ){
	    if( primes_dict.indexOf(c) != -1 ){
		primes.push(c);
	    }
	    else{
		comps.push(c);
	    }
	}
    }
    else{
	if( alpha.length < 25 ){
	    alpha.push(String.fromCharCode(c.charCodeAt(0) +1 ));
	}
    }
}
var p = 0;
var c = 0;
for(var i = 0; i < primes.length; i++){
    p += primes[i];
}

for(var i = 0; i < comps.length; i++){
    c += comps[i];
}

var numerics = p*c;
$("input[name=solution]").val(alpha.join('') + numerics);
$("input:last").click();